### PR TITLE
Add Radians::try_from_degrees

### LINKED ISSUE #13: Add Radians::try_from_degrees()

### PR DESCRIPTION
**Pull request by Mafii**
Resolves #13

### CODE REVIEW

File: src/radians.rs [hunk 1]
```diff
@@ -34,6 +34,11 @@ impl Radians {
     pub fn value(self) -> f64 {
         self.value
     }
+
+    /// Try getting the radians value of the degrees
+    pub fn try_from_degrees(degrees: f64) -> Result<Radians, RadiansError> {
```
Comment (jeremystucki) [inline src/radians.rs]: Use `Self` in the return type

File: src/radians.rs [hunk 2]
```diff
@@ -88,4 +93,62 @@ mod tests {
...
+        let degrees = 0.0;
+        let radians = Radians::try_from_degrees(degrees).unwrap();
+        let expected = 0.0;
+        assert_nearly_eq!(expected, radians.value())
+    }
+    #[test]
+    fn try_get_radians_works_with_360_as_input() {
+        let degrees = 360.0;
+        let radians = Radians::try_from_degrees(degrees);
+        match radians {
```
Comment (jeremystucki) [inline src/radians.rs]: Use `assert!(radians.is_err())`

File: src/radians.rs [hunk 3]
```diff
@@ -88,4 +93,62 @@ mod tests {
...
+        let degrees = 359.0;
+        let radians = Radians::try_from_degrees(degrees).unwrap();
+        let expected = 2.0 * PI - PI / 180.0;
+        assert_nearly_eq!(expected, radians.value())
+    }
+    #[test]
+    fn try_get_radians_returns_none_with_negative_1_as_input() {
+        let degrees = -1.0;
+        let radians = Radians::try_from_degrees(degrees);
+        match radians {
```
Comment (jeremystucki) [inline src/radians.rs]: See above

File: src/radians.rs [hunk 4]
```diff
@@ -88,4 +93,62 @@ mod tests {
...
+            Ok(_) => assert!(false, "Negative one isn not an allowed degrees value"),
+            Err(_) => assert!(true),
+        }
+    }
+    #[test]
+    fn try_get_radians_returns_none_with_361_as_input() {
+        let degrees = 361.0;
+        let radians = Radians::try_from_degrees(degrees);
+        match radians {
```
Comment (jeremystucki) [inline src/radians.rs]: See above

File: src/radians.rs [hunk 5]
```diff
@@ -88,4 +93,62 @@ mod tests {
...
+            Ok(_) => assert!(false, "360 degrees is too big. 2 PI as radiant is disallowed"),
+            Err(_) => assert!(true),
+        }
+    }
+    #[test]
+    fn try_get_radians_returns_none_with_359_as_input() {
+        let degrees = 359.0;
+        let radians = Radians::try_from_degrees(degrees).unwrap();
+        let expected = 2.0 * PI - PI / 180.0;
```
Comment (jeremystucki) [inline src/radians.rs]: Should we use braces to clarify the equation?

**Review by jeremystucki — CHANGES_REQUESTED** [review]

File: src/radians.rs (hunk 1 above)
Comment (jeremystucki) [reply src/radians.rs]: e.g. `Result<Self, RadiansError>`

File: src/radians.rs [hunk 6]
```diff
@@ -88,4 +93,62 @@ mod tests {
         assert_nearly_eq!(value, radians.value())
     }
 
+    #[test]
+    fn try_get_radians_works_with_0_as_input() {
+        let degrees = 0.0;
+        let radians = Radians::try_from_degrees(degrees).unwrap();
+        let expected = 0.0;
+        assert_nearly_eq!(expected, radians.value())
+    }
```
Comment (tautropfli) [inline src/radians.rs:119]: Missing empty line between fns

File: src/radians.rs (hunk 1 above)
Comment (jeremystucki) [inline src/radians.rs]: Should we call it `try_new_from_degrees`?

**Review by tautropfli — DISMISSED** [review]

File: src/radians.rs (hunk 1 above)
Comment (janhohenheim) [reply src/radians.rs]: `from_degrees`, follow the [convention of the standard library](https://doc.rust-lang.org/nightly/std/string/struct.String.html#method.from_utf8)

Comment (tautropfli) [reply src/radians.rs]: I think `from` and `new` are mutually exclusive in a function name…

File: src/radians.rs [hunk 7]
```diff
@@ -34,6 +34,11 @@ impl Radians {
     pub fn value(self) -> f64 {
         self.value
     }
+
+    /// Try getting the radians value of the degrees
```
Comment (janhohenheim) [inline src/radians.rs]: > Convert degrees to radians
> 
> ## Errors
> 
> ...
> 
> ## Examples
> 
> ...

File: src/radians.rs (hunk 1 above)
Comment (Mafii) [reply src/radians.rs]: `from_degrees` seems sensible to me.

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Add your changes to the changelog as well

File: src/radians.rs (hunk 1 above)
Comment (tautropfli) [reply src/radians.rs]: Aw fish! You were faster @jnferner.

**Review by jeremystucki — DISMISSED** [review]

File: src/radians.rs [hunk 8]
```diff
@@ -35,8 +35,8 @@ impl Radians {
         self.value
     }
 
-    /// Try getting the radians value of the degrees
-    pub fn try_from_degrees(degrees: f64) -> Result<Radians, RadiansError> {
+    /// Convert degrees to radians
```
Comment (janhohenheim) [inline src/radians.rs:38]: Add `## Errors` and `## Examples` sections

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/radians.rs [hunk 9]
```diff
@@ -34,6 +34,11 @@ impl Radians {
     pub fn value(self) -> f64 {
         self.value
     }
+
+    /// Convert degrees to radians
+    pub fn from_degrees(degrees: f64) -> Result<Self, RadiansError> {
```
Comment (janhohenheim) [inline src/radians.rs]: After internal discussion, me and @bash settled on `try_from_degrees`

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/radians.rs [hunk 10]
```diff
@@ -34,6 +34,11 @@ impl Radians {
...
+        return Radians::try_new(degrees / 360.0 * 2.0 * PI);
```
Comment (janhohenheim) [inline src/radians.rs]: How about
```
const MAX_DEGREES: f64 = 360.0;
const MAX_RADIANS: f64 = 2.0 * PI;
```
?

File: src/radians.rs (hunk 9 above)
Comment (Mafii) [reply src/radians.rs]: Resolved in 1a875aa90ffead22e502431154fdccd7e47ab25a

File: src/radians.rs (hunk 10 above)
Comment (janhohenheim) [reply src/radians.rs]: Remove that `return`

File: src/radians.rs [hunk 11]
```diff
@@ -88,4 +93,56 @@ mod tests {
         assert_nearly_eq!(value, radians.value())
     }
 
+    #[test]
+    fn try_get_radians_works_with_0_as_input() {
```
Comment (janhohenheim) [inline src/radians.rs]: The name of these tests is outdated

**Comment by Mafii** [comment]
> Add your changes to the changelog as well

Done

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/radians.rs (hunk 11 above)
Comment (jnferner) [reply src/radians.rs]: Could you write `0` as `zero` for consistency?

File: src/radians.rs (hunk 10 above)
Comment (Mafii) [reply src/radians.rs]: Resolved in  ae71de9b427dc409ad78b6f205b4e5ff28332e40

File: src/radians.rs (hunk 11 above)
Comment (Mafii) [reply src/radians.rs]: Resolved in e69c0a649d6a3f0041fb3f3597a49b9dbda1c98e

File: src/radians.rs (hunk 8 above)
Comment (Mafii) [reply src/radians.rs:38]: Resolved in 3569e7691b9d91c2eb7d6762d6ccc76d487d8f01

Comment (janhohenheim) [reply src/radians.rs:38]: Use [std::f64::consts::FRAC_PI_2](https://doc.rust-lang.org/std/f64/consts/constant.FRAC_PI_2.html) instead of `PI / 2.0`

**Review by janhohenheim — CHANGES_REQUESTED** [review]

**Comment by Mafii** [comment]
Force push to fix typo: Property should've been properly

**Review by janhohenheim — DISMISSED** [review]

File: changelog.md [hunk 12]
```diff
@@ -22,3 +22,4 @@
 
 ## TBD
 - Add `Vector::negative`
+- Add `Radians::from_degree`
```
Comment (jeremystucki) [inline changelog.md]: This is not the right function name

**Review by jeremystucki — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 12 above)
Comment (Mafii) [reply changelog.md]: Resolved in d6928345667f35b63543316836c535f1804fa94a

**Review by jeremystucki — APPROVED** [review]

**Pull request merged**